test/BrainPass.t.sol

```diff
@@ -0,0 +1,240 @@
+        vm.stopPrank();
+    }
+
+    function testMintDurationNotInTimeFrame() public {
+        mockERC20.mint(alice, 20000e18);
+        vm.startPrank(alice);
+        mockERC20.approve(address(BrainPass), 9000e18);
+        vm.expectRevert(BrainPassCollectibles.DurationNotInTimeFrame.selector);
+        BrainPass.mintNFT(1, 172800, 518400);
+        vm.stopPrank();
+    }
+
+    function testInvalidMaxTokensForAPass() public {
+        vm.expectRevert(
+            BrainPassCollectibles.InvalidMaxTokensForAPass.selector
+        );
+        BrainPass.addPassType(
+            15e18,
+            "http://example.orgs",
+            "OleanjiPass",
+            0,
+            0
+        );
+    }
+
+    function testPassTypeNotFound() public {
+        mockERC20.mint(alice, 20000e18);
+        vm.startPrank(alice);
+        mockERC20.approve(address(BrainPass), 9000e18);
+        vm.expectRevert(BrainPassCollectibles.PassTypeNotFound.selector);
+        BrainPass.mintNFT(4, 172800, 5184000);
+    }
+
+    function testCannotMintPausedPass() public {
+        BrainPass.togglePassTypeStatus(1);
+        mockERC20.mint(alice, 20000e18);
+        vm.startPrank(alice);
+        mockERC20.approve(address(BrainPass), 9000e18);
+        vm.expectRevert(
+            BrainPassCollectibles.CannotMintPausedPassType.selector
+        );
+        BrainPass.mintNFT(1, 172800, 5184000);
+    }
+
+    function testmintNFT() public {
+        mockERC20.mint(alice, 20000e18);
+        vm.startPrank(alice);
+        mockERC20.approve(address(BrainPass), 3e18);
+        vm.expectRevert(stdError.arithmeticError);
+        BrainPass.mintNFT(1, 172800, 5184000);
+        mockERC20.approve(address(BrainPass), 9000e18);
+        assertEq(BrainPass.balanceOf(alice), 0);
+        BrainPass.mintNFT(1, 172800, 5184000);
+        assertEq(BrainPass.balanceOf(alice), 1);
+        assertEq(mockERC20.balanceOf(address(this)), 870e18);
+        uint256 mintedPass = BrainPass.getUserPassDetails(alice).tokenId;
+        assertEq(mintedPass, 1);
+        vm.stopPrank();
+    }
+
+    function testDifferentPassMint() public {
+        BrainPass.addPassType(
+            15e18,
+            "http://example.orgs",
+            "OleanjiPass",
+            2,
+            0
+        );
+        BrainPass.addPassType(15e18, "http://oleanji.com", "KesarPass", 2, 0);
+        mockERC20.mint(alice, 20000e18);
+        mockERC20.mint(doe, 20000e18);
+
+        mockERC20.mint(bob, 20000e18);
+
+        vm.startPrank(alice);
+        mockERC20.approve(address(BrainPass), 20000e18);
+        BrainPass.mintNFT(1, 172800, 5184000); //"http://example.com"
+        uint256 firstId = BrainPass.getUserPassDetails(alice).tokenId;
+        console.log(BrainPass.tokenURI(firstId), firstId);
+        vm.stopPrank();
+        vm.startPrank(doe);
+        mockERC20.approve(address(BrainPass), 20000e18);
+        BrainPass.mintNFT(2, 172800, 5184000); // "http://example.orgs",
```

Comment:
in those tests you are not testing that urls are the right ones as I told you. you are doing some console logs but test doesnt check it